### PR TITLE
Chore(New_Exp): Add stress chaos experiments and split out the exec stress experiments

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: uditgaurav <udit@chaosnative.com>

## This PR includes:

- New Stress experiments using non-exec mode: This includes the addition of a new set of stress chaos experiments using `cgroups` and `nsutil`.
- The included experiments are:
  - pod-cpu-hog
  - pod-memory-hog
  - pod-io-stress
- Splitting out the exec stress chaos experiment: Rename the older version of the stress experiment with the exec in suffix.
- The experiment has been tested on `docker` and `containerd` cluster runtime.

### CODE REVIEW

**Comment by uditgaurav** [comment]
## Process on the target container
- _During Chaos Injection_

```bash
root@nginx-86c57db685-gdsng:/# ps aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.0  0.0  10648  6012 ?        Ss   19:37   0:00 nginx: master process nginx -g daemon off;
nginx       31  0.0  0.0  11052  2648 ?        S    19:37   0:00 nginx: worker process
root        34  0.0  0.0   3864  3320 pts/0    Ss   19:49   0:00 bash
root       367  0.0  0.0  50604  4244 ?        SL   20:01   0:00 stress-ng --timeout 60s --cpu 1
root       368 98.5  0.0  50604  4816 ?        R    20:01   0:31 stress-ng-cpu
root       369  0.0  0.0   7636  2720 pts/0    R+   20:01   0:00 ps aux
```

- _After Chaos Completion_

```bash
root@nginx-86c57db685-gdsng:/# ps aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.0  0.0  10648  6012 ?        Ss   19:37   0:00 nginx: master process nginx -g daemon off;
nginx       31  0.0  0.0  11052  2648 ?        S    19:37   0:00 nginx: worker process
root        34  0.0  0.0   3864  3320 pts/0    Ss   19:49   0:00 bash
root       370  0.0  0.0   7636  2756 pts/0    R+   20:02   0:00 ps aux
```

## Kubectl Top Output

```bash
Every 2.0s: kubectl top pods   
                                                     
NAME                        CPU(cores)   MEMORY(bytes)
nginx-86c57db685-gdsng      965m         31Mi
nginx-chaos-runner          3m           9Mi
pod-cpu-hog-helper-tsvmba   8m           8Mi
pod-cpu-hog-zu6qwa-26z29    2m           10Mi
```

## Helper Pod Logs
```bash
udit@ubuntu ~/g/s/g/l/litmus-go> kubectl logs -f pod-cpu-hog-helper-axyyul
W0520 20:32:00.226591   10306 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2021-05-20T20:32:00Z" level=info msg="[PreReq]: Getting the ENV variables"
time="2021-05-20T20:32:00Z" level=info msg="containerid: 30e9e141a51c"
time="2021-05-20T20:32:00Z" level=info msg="[Info]: Container ID=30e9e141a51c has process PID=4043"
time="2021-05-20T20:32:00Z" level=info msg="[Info]: Details of Stressor:" CPU Core=1 Timeout=60
time="2021-05-20T20:32:00Z" level=info msg="[Info]: starting process: pause nsutil -t 4043 -p -- stress-ng --timeout 60s --cpu 1"
time="2021-05-20T20:32:00Z" level=info msg="[Info]: Sending signal to resume the stress process"
time="2021-05-20T20:32:01Z" level=info msg="[Wait]: Waiting for chaos completion"
time="2021-05-20T20:33:01Z" level=info msg="[Info]: Chaos injection completed"
time="2021-05-20T20:33:01Z" level=info msg="[Info]: Stress process removed sucessfully"
```

## Experiment Pod Logs

```bash
udit@ubuntu ~/g/s/g/l/s/litmus-go> kubectl logs -f pod-cpu-hog-zu6qwa-26z29
W0520 20:01:15.070845       1 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2021-05-20T20:01:15Z" level=info msg="Experiment Name: pod-cpu-hog"
time="2021-05-20T20:01:15Z" level=info msg="[PreReq]: Getting the ENV for the  experiment"
time="2021-05-20T20:01:15Z" level=info msg="[PreReq]: Updating the chaos result of pod-cpu-hog experiment (SOT)"
time="2021-05-20T20:01:15Z" level=info msg="The application information is as follows" Ramp Time=0 Namespace=default Label="app=nginx" Chaos Duration=60
time="2021-05-20T20:01:15Z" level=info msg="[Status]: Verify that the AUT (Application Under Test) is running (pre-chaos)"
time="2021-05-20T20:01:15Z" level=info msg="[Status]: Checking whether application containers are in ready state"
time="2021-05-20T20:01:15Z" level=info msg="[Status]: The Container status are as follows" Readiness=true container=nginx Pod=nginx-86c57db685-gdsng
time="2021-05-20T20:01:17Z" level=info msg="[Status]: Checking whether application pods are in running state"
time="2021-05-20T20:01:17Z" level=info msg="[Status]: The status of Pods are as follows" Status=Running Pod=nginx-86c57db685-gdsng
time="2021-05-20T20:01:19Z" level=info msg="[Chaos]:Number of pods targeted: 1"
time="2021-05-20T20:01:19Z" level=info msg="[Info]: Target pods list for chaos, [nginx-86c57db685-gdsng]"
time="2021-05-20T20:01:19Z" level=info msg="[Info]: Details of application under chaos injection" PodName=nginx-86c57db685-gdsng NodeName=ip-192-168-66-49.us-west-2.compute.internal ContainerName=nginx
time="2021-05-20T20:01:19Z" level=info msg="[Status]: Checking the status of the helper pods"
time="2021-05-20T20:01:19Z" level=info msg="[Status]: Checking whether application containers are in ready state"
time="2021-05-20T20:01:21Z" level=info msg="[Status]: Checking whether application pods are in running state"
time="2021-05-20T20:01:23Z" level=info msg="[Status]: The status of Pods are as follows" Pod=pod-cpu-hog-helper-tsvmba Status=Running
time="2021-05-20T20:01:25Z" level=info msg="[Wait]: waiting till the completion of the helper pod"
time="2021-05-20T20:01:25Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:26Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:27Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:28Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:29Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:30Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:31Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:32Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:33Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:34Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:35Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:36Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:37Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:38Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:39Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:40Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:41Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:42Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:43Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:44Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:45Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:46Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:47Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:48Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:49Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:50Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:51Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:52Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:53Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:54Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:55Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:56Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:57Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:58Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:01:59Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:00Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:01Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:02Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:03Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:04Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:05Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:06Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:07Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:08Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:09Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:10Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:11Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:12Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:13Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:14Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:15Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:16Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:17Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:18Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:19Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:20Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:21Z" level=info msg="helper pod status: Running"
time="2021-05-20T20:02:22Z" level=info msg="helper pod status: Succeeded"
time="2021-05-20T20:02:22Z" level=info msg="[Status]: The running status of Pods are as follows" Pod=pod-cpu-hog-helper-tsvmba Status=Succeeded
time="2021-05-20T20:02:23Z" level=info msg="[Cleanup]: Deleting all the helper pod"
time="2021-05-20T20:02:25Z" level=info msg="[Confirmation]: pod-cpu-hog chaos has been injected successfully"
time="2021-05-20T20:02:25Z" level=info msg="[Status]: Verify that the AUT (Application Under Test) is running (post-chaos)"
time="2021-05-20T20:02:25Z" level=info msg="[Status]: Checking whether application containers are in ready state"
time="2021-05-20T20:02:25Z" level=info msg="[Status]: The Container status are as follows" container=nginx Pod=nginx-86c57db685-gdsng Readiness=true
time="2021-05-20T20:02:27Z" level=info msg="[Status]: Checking whether application pods are in running state"
time="2021-05-20T20:02:27Z" level=info msg="[Status]: The status of Pods are as follows" Pod=nginx-86c57db685-gdsng Status=Running
time="2021-05-20T20:02:29Z" level=info msg="[The End]: Updating the chaos result of pod-cpu-hog experiment (EOT)"
```

File: chaoslib/litmus/stress-chaos/helper/stress-helper.go [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+}
+
+//getErrorPath will give the invalid cgroup path
+func getErrorPath(err error) cgroups.Path {
+	return func(_ cgroups.Name) (string, error) {
+		return "", err
+	}
+}
+
+func getCgroupDestination(pid int, subsystem string) (string, error) {
```
Comment (ispeakc0de) [inline chaoslib/litmus/stress-chaos/helper/stress-helper.go:437]: can we add comment here?

File: chaoslib/litmus/stress-chaos/helper/stress-helper.go [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+	mountinfoPath := fmt.Sprintf("/proc/%d/mountinfo", pid)
```
Comment (ispeakc0de) [inline chaoslib/litmus/stress-chaos/helper/stress-helper.go:438]:
```suggestion
	mountInfoPath := fmt.Sprintf("/proc/%d/mountinfo", pid)
```

File: chaoslib/litmus/stress-chaos/helper/stress-helper.go [hunk 3]
```diff
@@ -0,0 +1,560 @@
...
+		for _, opt := range strings.Split(fields[len(fields)-1], ",") {
+			if opt == subsystem {
+				return fields[3], nil
+			}
+		}
+	}
+	if err := s.Err(); err != nil {
+		return "", err
+	}
+	return "", errors.Errorf("never found desct for %v ", subsystem)
```
Comment (ispeakc0de) [inline chaoslib/litmus/stress-chaos/helper/stress-helper.go:456]:
```suggestion
	return "", errors.Errorf("no destination found for %v ", subsystem)
```

File: chaoslib/litmus/stress-chaos/lib/stress-chaos.go [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+		log.InfoWithValues("[Info]: Details of application under chaos injection", logrus.Fields{
+			"PodName":       pod.Name,
+			"NodeName":      pod.Spec.NodeName,
+			"ContainerName": experimentsDetails.TargetContainer,
+		})
+		runID := common.GetRunID()
+		err := createHelperPod(experimentsDetails, clients, pod.Name, pod.Spec.NodeName, runID, labelSuffix)
+		if err != nil {
+			return errors.Errorf("unable to create the helper pod, err: %v", err)
+		}
```
Comment (ispeakc0de) [inline chaoslib/litmus/stress-chaos/lib/stress-chaos.go:107]:
```suggestion
		if err := createHelperPod(experimentsDetails, clients, pod.Name, pod.Spec.NodeName, runID, labelSuffix);err != nil {
			return errors.Errorf("unable to create the helper pod, err: %v", err)
		}
```

File: chaoslib/litmus/stress-chaos/lib/stress-chaos.go [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+
+		appLabel := "name=" + experimentsDetails.ExperimentName + "-helper-" + runID
+
+		//checking the status of the helper pods, wait till the pod comes to running state else fail the experiment
+		log.Info("[Status]: Checking the status of the helper pods")
+		err = status.CheckApplicationStatus(experimentsDetails.ChaosNamespace, appLabel, experimentsDetails.Timeout, experimentsDetails.Delay, clients)
+		if err != nil {
+			common.DeleteHelperPodBasedOnJobCleanupPolicy(experimentsDetails.ExperimentName+"-helper-"+runID, appLabel, chaosDetails, clients)
+			return errors.Errorf("helper pods are not in running state, err: %v", err)
+		}
```
Comment (ispeakc0de) [inline chaoslib/litmus/stress-chaos/lib/stress-chaos.go:117]: Can we use the `status.CheckHelperStatus` func here?

File: chaoslib/litmus/stress-chaos/lib/stress-chaos.go [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+		if err != nil || podStatus == "Failed" {
+			common.DeleteHelperPodBasedOnJobCleanupPolicy(experimentsDetails.ExperimentName+"-helper-"+runID, appLabel, chaosDetails, clients)
+			return errors.Errorf("helper pod failed due to, err: %v", err)
+		}
+
+		//Deleting all the helper pod for stress chaos
+		log.Info("[Cleanup]: Deleting the helper pod")
+		err = common.DeletePod(experimentsDetails.ExperimentName+"-helper-"+runID, appLabel, experimentsDetails.ChaosNamespace, chaosDetails.Timeout, chaosDetails.Delay, clients)
+		if err != nil {
+			return errors.Errorf("Unable to delete the helper pods, err: %v", err)
```
Comment (ispeakc0de) [inline chaoslib/litmus/stress-chaos/lib/stress-chaos.go:132]: Error inside errors.Errorf() func should start with a lower case alphabet

File: chaoslib/litmus/stress-chaos/lib/stress-chaos.go [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+		if err != nil {
+			return errors.Errorf("unable to create the helper pod, err: %v", err)
+		}
+	}
+
+	appLabel := "app=" + experimentsDetails.ExperimentName + "-helper-" + labelSuffix
+
+	//checking the status of the helper pods, wait till the pod comes to running state else fail the experiment
+	log.Info("[Status]: Checking the status of the helper pods")
+	err := status.CheckApplicationStatus(experimentsDetails.ChaosNamespace, appLabel, experimentsDetails.Timeout, experimentsDetails.Delay, clients)
```
Comment (ispeakc0de) [inline chaoslib/litmus/stress-chaos/lib/stress-chaos.go:170]: same as above

File: experiments/generic/pod-cpu-hog-exec/experiment/pod-cpu-hog-exec.go [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+	// Including the litmus lib for pod-cpu-hog-exec
+	switch experimentsDetails.ChaosLib {
+	case "litmus":
+		if err := litmusLIB.PrepareCPUExecStress(&experimentsDetails, clients, &resultDetails, &eventsDetails, &chaosDetails); err != nil {
+			log.Errorf("[Error]: CPU hog failed, err: %v", err)
+			failStep := "failed in chaos injection phase"
+			result.RecordAfterFailure(&chaosDetails, &resultDetails, failStep, clients, &eventsDetails)
+			return
+		}
+	case "pumba":
```
Comment (ispeakc0de) [inline experiments/generic/pod-cpu-hog-exec/experiment/pod-cpu-hog-exec.go:115]: Do we want to add pumba lib here? as it is not an exec chaos

File: experiments/generic/pod-cpu-hog-exec/rbac.yaml [hunk 9]
```diff
@@ -0,0 +1,36 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: pod-cpu-hog-sa
```
Comment (ispeakc0de) [inline experiments/generic/pod-cpu-hog-exec/rbac.yaml:5]: do we want to rename the sa name also?

File: experiments/generic/pod-cpu-hog/README.md [hunk 10]
```diff
@@ -8,7 +8,7 @@
 </tr>
 <tr>
  <td> Pod CPU Hog </td>
- <td> This experiment causes CPU resource consumption on specified application containers by starting one or more md5sum calculation process on the special file /dev/zero. It Can test the application's resilience to potential slowness/unavailability of some replicas due to high CPU load.</td>
+ <td> </td>
```
Comment (ispeakc0de) [inline experiments/generic/pod-cpu-hog/README.md:11]: we can some details here

File: experiments/generic/pod-cpu-hog/experiment/pod-cpu-hog.go [hunk 11]
```diff
@@ -106,14 +105,7 @@ func PodCPUHog(clients clients.ClientSets) {
...
 	case "litmus":
-		if err := litmusLIB.PrepareCPUstress(&experimentsDetails, clients, &resultDetails, &eventsDetails, &chaosDetails); err != nil {
-			log.Errorf("[Error]: CPU hog failed, err: %v", err)
-			failStep := "failed in chaos injection phase"
-			result.RecordAfterFailure(&chaosDetails, &resultDetails, failStep, clients, &eventsDetails)
-			return
-		}
-	case "pumba":
-		if err := pumbaLIB.PreparePodCPUHog(&experimentsDetails, clients, &resultDetails, &eventsDetails, &chaosDetails); err != nil {
+		if err := litmusLIB.PrepareAndInjectStressChaos(&experimentsDetails, clients, &resultDetails, &eventsDetails, &chaosDetails); err != nil {
```
Comment (ispeakc0de) [inline experiments/generic/pod-cpu-hog/experiment/pod-cpu-hog.go:108]: Is should also support pumba lib right?

File: experiments/generic/pod-memory-hog-exec/experiment/pod-memory-hog-exec.go [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+	// Including the litmus lib for pod-memory-hog-exec
+	switch experimentsDetails.ChaosLib {
+	case "litmus":
+		if err := litmusLIB.PrepareMemoryExecStress(&experimentsDetails, clients, &resultDetails, &eventsDetails, &chaosDetails); err != nil {
+			log.Errorf("[Error]: pod memory hog failed, err: %v", err)
+			failStep := "failed in chaos injection phase"
+			result.RecordAfterFailure(&chaosDetails, &resultDetails, failStep, clients, &eventsDetails)
+			return
+		}
+	case "pumba":
```
Comment (ispeakc0de) [inline experiments/generic/pod-memory-hog-exec/experiment/pod-memory-hog-exec.go:116]: same as above

File: experiments/generic/pod-memory-hog/README.md [hunk 13]
```diff
@@ -8,7 +8,7 @@
 </tr>
 <tr>
  <td> Pod Memory Hog </td>
- <td> This experiment causes Memory resource consumption on specified application containers by using dd command which will used to consume memory of the application container for certain duration of time. It can test the application's resilience to potential slowness/unavailability of some replicas due to high Memory load.</td>
+ <td> </td>
```
Comment (ispeakc0de) [inline experiments/generic/pod-memory-hog/README.md:11]: same as above

File: pkg/generic/stress-chaos/environment/environment.go [hunk 14]
```diff
@@ -22,20 +22,25 @@ func GetENV(experimentDetails *experimentTypes.ExperimentDetails) {
...
 	experimentDetails.TargetPods = common.Getenv("TARGET_PODS", "")
-	experimentDetails.NumberOfWorkers, _ = strconv.Atoi(common.Getenv("NUMBER_OF_WORKERS", "4"))
-	experimentDetails.LIBImage = common.Getenv("LIB_IMAGE", "litmuschaos/go-runner:latest")
-	experimentDetails.LIBImagePullPolicy = common.Getenv("LIB_IMAGE_PULL_POLICY", "Always")
 	experimentDetails.PodsAffectedPerc, _ = strconv.Atoi(common.Getenv("PODS_AFFECTED_PERC", "0"))
-	experimentDetails.Sequence = common.Getenv("SEQUENCE", "parallel")
+	experimentDetails.ContainerRuntime = common.Getenv("CONTAINER_RUNTIME", "docker")
+	experimentDetails.ChaosServiceAccount = common.Getenv("CHAOS_SERVICE_ACCOUNT", "")
+	experimentDetails.CPUcores, _ = strconv.Atoi(common.Getenv("CPU_CORES", "1"))
+	experimentDetails.FilesystemUtilizationPercentage, _ = strconv.Atoi(common.Getenv("FILESYSTEM_UTILIZATION_PERCENTAGE", ""))
```
Comment (ispeakc0de) [inline pkg/generic/stress-chaos/environment/environment.go:36]: Can we do something similar to [this](https://github.com/litmuschaos/litmus-go/blob/d8125aef2470b140eed4870a6f59d721c49209aa/pkg/aws-ssm/aws-ssm-chaos/environment/environment.go#L43)

**Review by ispeakc0de — CHANGES_REQUESTED** [review]

File: chaoslib/litmus/stress-chaos/helper/stress-helper.go [hunk 15]
```diff
@@ -0,0 +1,560 @@
...
+	return nil
+}
+
+//terminateProcess will remove the stress process from the target container after chaos completion
+func terminateProcess(pid int) error {
+	process, err := os.FindProcess(pid)
+	if err != nil {
+		return errors.Errorf("unreachable path, err: %v", err)
+	}
+	if err = process.Signal(syscall.SIGTERM); err != nil && err.Error() != "os: process already finished" {
```
[the remark this answers is not in the document]
Comment (uditgaurav) [reply chaoslib/litmus/stress-chaos/helper/stress-helper.go]: It does not require `string.Contains()` function, have tested it. Yes we can give constants.

File: chaoslib/litmus/stress-chaos/helper/stress-helper.go (hunk 2 above)
Comment (uditgaurav) [reply chaoslib/litmus/stress-chaos/helper/stress-helper.go:404]: This is fine. The file name is `mountinfoPath` inside the container.

File: chaoslib/litmus/stress-chaos/helper/stress-helper.go [hunk 16]
```diff
@@ -0,0 +1,560 @@
...
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, errors.Errorf("unable to parse cgroup file: %v", err)
+	}
+	defer file.Close()
+	return parseCgroupFromReader(file)
+}
+
+func parseCgroupFromReader(r io.Reader) (map[string]string, error) {
+	var (
```
[the remark this answers is not in the document]
Comment (uditgaurav) [reply chaoslib/litmus/stress-chaos/helper/stress-helper.go:340]: No specific reason. There are multiple variables so we can keep them inside a block.

File: chaoslib/litmus/stress-chaos/helper/stress-helper.go [hunk 17]
```diff
@@ -0,0 +1,560 @@
...
+	if err := s.Err(); err != nil {
+		return nil, errors.Errorf("buffer scanner failed: %v", err)
+	}
+
+	return cgroups, nil
+}
+
+//getExistingPath will be used to get the existing valid cgroup path
+func getExistingPath(paths map[string]string, pid int, suffix string) cgroups.Path {
+	for n, p := range paths {
```
[the remark this answers is not in the document]
Comment (uditgaurav) [reply chaoslib/litmus/stress-chaos/helper/stress-helper.go:367]: namespace and path

File: chaoslib/litmus/stress-chaos/lib/stress-chaos.go (hunk 7 above)
Comment (uditgaurav) [reply chaoslib/litmus/stress-chaos/lib/stress-chaos.go]: This should be mandatory. We could add a lint to check this.

File: chaoslib/litmus/stress-chaos/helper/stress-helper.go (hunk 17 above)
[the remark this answers is not in the document]
Comment (ispeakc0de) [reply chaoslib/litmus/stress-chaos/helper/stress-helper.go:367]: Can we use the full name?

File: chaoslib/litmus/stress-chaos/lib/stress-chaos.go (hunk 7 above)
Comment (ispeakc0de) [reply chaoslib/litmus/stress-chaos/lib/stress-chaos.go]: I mean we need to add this func `status.CheckHelperStatus` instead? not sure if go-lint can help to suggest this

File: chaoslib/litmus/stress-chaos/helper/stress-helper.go (hunk 15 above)
[the remark this answers is not in the document]
Comment (ispeakc0de) [reply chaoslib/litmus/stress-chaos/helper/stress-helper.go]: Okay we can keep inequality operator if you are confident about the exact error string

**Review by ispeakc0de — APPROVED** [review]

**Review by ksatchit — APPROVED** [review]

**Pull request merged**